File: SIH26011/backend/app/api/audit.py
from fastapi import APIRouter
from typing import List
from app.repositories.data_repository import repository
from app.schemas.pydantic_schemas import AuditEventResponse, RevisionResponse

router = APIRouter(prefix="/api/audit", tags=["Audit & Revisions"])

@router.get("", response_model=List[AuditEventResponse])
def get_audit_log():
    return [AuditEventResponse(**e.__dict__) for e in repository.audit_log]

@router.get("/revisions", response_model=List[RevisionResponse])
def get_revisions():
    return [RevisionResponse(**r.__dict__) for r in repository.revisions]

File: backend/src/routes/audit.js
import { Router } from "express";
import { repository } from "../repository.js";

const router = Router();

// Get complete audit log
router.get("/", (_req, res) => {
  res.json(repository.audit_log);
});

// Get revision history
router.get("/revisions", (_req, res) => {
  res.json(repository.revisions);
});

export default router;